tests/e2e/driver/driver.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package driver
+
+import (
+	v1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type PVTestDriver interface {
+	DynamicPVTestDriver
+	PreProvisionedVolumeTestDriver
+}
+
+// DynamicPVTestDriver represents an interface for a CSI driver that supports DynamicPV
+type DynamicPVTestDriver interface {
+	// GetDynamicProvisionStorageClass returns a StorageClass dynamic provision Persistent Volume
+	GetDynamicProvisionStorageClass(parameters map[string]string, mountOptions []string, reclaimPolicy *v1.PersistentVolumeReclaimPolicy, volumeExpansion *bool, bindingMode *storagev1.VolumeBindingMode, allowedTopologyValues []string, namespace string) *storagev1.StorageClass
+}
+
+// PreProvisionedVolumeTestDriver represents an interface for a CSI driver that supports pre-provisioned volume
+type PreProvisionedVolumeTestDriver interface {
+	// GetPersistentVolume returns a PersistentVolume with pre-provisioned volumeHandle
+	GetPersistentVolume(volumeID string, fsType string, size string, reclaimPolicy *v1.PersistentVolumeReclaimPolicy, namespace string) *v1.PersistentVolume
+}
+
+func getStorageClass(
+	generateName string,
+	provisioner string,
+	parameters map[string]string,
+	mountOptions []string,
+	reclaimPolicy *v1.PersistentVolumeReclaimPolicy,
+	volumeExpansion *bool,
+	bindingMode *storagev1.VolumeBindingMode,
+	allowedTopologies []v1.TopologySelectorTerm,
+) *storagev1.StorageClass {
+	if reclaimPolicy == nil {
+		defaultReclaimPolicy := v1.PersistentVolumeReclaimDelete
+		reclaimPolicy = &defaultReclaimPolicy
```

Comment:
see if this `StringPtr` function can be used here from https://github.com/kubernetes/utils/blob/6203023598ed91944d54e72f57454dc6da9a4e17/pointer/pointer.go#L162 instead of an additional variable declaration here.

Reply:
`v1.PersistentVolumeReclaimDelete` is of type `v1.PersistentVolumeReclaimPolicy`
using StringPtr requires additional type conversions
```
reclaimPolicy = (*v1.PersistentVolumeReclaimPolicy)(pointer_util.StringPtr(string(v1.PersistentVolumeReclaimDelete))) 
```

I feel the existing code is simple and clear.
What do you prefer?

Reply:
I see that, lets keep the current implementation.